### PR TITLE
Replaces cosmic heretic clone damage with organ damage

### PR DESCRIPTION
**Pull request by strandsofivy**
## About The Pull Request

This PR replaces clone damage dealt by the cosmic blade with damage to a random organ (8, 12, 14 respectively). To compensate for this, burn damage will now deal 5-14-28, which is slightly weaker immediate damage than before (6-16-32).

Someone suggested that blades should remove blood from the victim, however I felt like that was too punishing considering the speed of blood regeneration.

Feel free to share suggestions in the comments, if you know of a better idea. Main point of this PR is to get rid of clone damage sources.
## Why It's Good For The Game

These are the last sources of clone damage in the game. We don't need it anymore.
## Changelog
:cl:
balance: Clone damage dealt by the cosmic blade has been replaced with organ damage and increased burn damage. Clone damage dealt by the cosmic beam has been removed. The star gazer now deals burn damage instead of clone damage.
fix: The health of mobs combo'd by a cosmic blade will now update correctly.
/:cl:

### CODE REVIEW

**Review by MrMelbert — COMMENTED** [review]
A little disappointing because Rust also does organ damage but I don't have other ideas

File: code/modules/antagonists/heretic/knowledge/cosmic_lore.dm [hunk 1]
```diff
@@ -171,18 +179,18 @@
 		combo_counter += 1
 	if(second_target_resolved)
 		new /obj/effect/temp_visual/cosmic_explosion(get_turf(second_target_resolved))
-		playsound(get_turf(second_target_resolved), 'sound/magic/cosmic_energy.ogg', 25, FALSE)
+		playsound(get_turf(second_target_resolved), 'sound/magic/cosmic_energy.ogg', 14, FALSE)
```
[the remark this answers is not in the document]
Comment (distributivgesetz) [reply code/modules/antagonists/heretic/knowledge/cosmic_lore.dm:182]: i
didnt meant o replace that

**Comment by MrMelbert** [comment]
Yeah but both flesh and knock both do blood loss (via wound) so it's a wash

File: code/modules/antagonists/heretic/knowledge/cosmic_lore.dm [hunk 2]
```diff
@@ -161,7 +169,7 @@
 	var/mob/living/third_target_resolved = third_target?.resolve()
 	var/need_mob_update = FALSE
 	need_mob_update += target.adjustFireLoss(4, updating_health = FALSE)
-	need_mob_update += target.adjustCloneLoss(2, updating_health = FALSE)
+	need_mob_update += target.adjustOrganLoss(pick(valid_organ_slots), 8)
 	if(need_mob_update)
 		target.updatehealth()
 	if(target == second_target_resolved || target == third_target_resolved)
```
Comment (timothymtorres) [inline code/modules/antagonists/heretic/knowledge/cosmic_lore.dm:175]: If the second or third target are different mobs, then why are we applying damage to them and then updating the original targets health?

File: code/modules/antagonists/heretic/knowledge/cosmic_lore.dm [hunk 3]
```diff
@@ -174,15 +182,15 @@
 		playsound(get_turf(second_target_resolved), 'sound/magic/cosmic_energy.ogg', 25, FALSE)
 		need_mob_update = FALSE
 		need_mob_update += second_target_resolved.adjustFireLoss(10, updating_health = FALSE)
-		need_mob_update += second_target_resolved.adjustCloneLoss(6, updating_health = FALSE)
+		need_mob_update += second_target_resolved.adjustOrganLoss(pick(valid_organ_slots), 14)
 		if(need_mob_update)
 			target.updatehealth()
```
Comment (timothymtorres) [inline code/modules/antagonists/heretic/knowledge/cosmic_lore.dm:187]:
```suggestion
		need_mob_update += second_target_resolved.adjustOrganLoss(pick(valid_organ_slots), 14)
		if(need_mob_update)
			second_target_resolved.updatehealth()
```
Wouldn't this be better?

File: code/modules/antagonists/heretic/knowledge/cosmic_lore.dm [hunk 4]
```diff
@@ -174,15 +182,15 @@
...
 		if(third_target_resolved)
 			new /obj/effect/temp_visual/cosmic_domain(get_turf(third_target_resolved))
 			playsound(get_turf(third_target_resolved), 'sound/magic/cosmic_energy.ogg', 50, FALSE)
 			need_mob_update = FALSE
 			need_mob_update += third_target_resolved.adjustFireLoss(20, updating_health = FALSE)
-			need_mob_update += third_target_resolved.adjustCloneLoss(12, updating_health = FALSE)
+			need_mob_update += third_target_resolved.adjustOrganLoss(pick(valid_organ_slots), 20)
 			if(need_mob_update)
 				target.updatehealth()
```
Comment (timothymtorres) [inline code/modules/antagonists/heretic/knowledge/cosmic_lore.dm:195]:
```suggestion
				third_target_resolved.updatehealth()
```

Same as above.

File: code/modules/antagonists/heretic/knowledge/cosmic_lore.dm (hunk 2 above)
Comment (distributivgesetz) [reply code/modules/antagonists/heretic/knowledge/cosmic_lore.dm:175]: Seems to be oversights from the previous refactor, I'll fix them tomorrow while I'm already at it

**Comment by distributivgesetz** [comment]
The lack of immediate damage is a nerf to cosmic blades, actually

**Comment by YuliceTheDruid** [comment]
> The lack of more immediate damage is a nerf to cosmic blades, actually now that you say that

It's a pretty hefty nerf to cosmic, both the lack of immediate damage and the fact that organ damage does FUCK ALL unless you completely kill the organ, and that organ damage slowly regenerates over time anyway. The only organ (afaik) that's actually affected by damage beyond death is the brain. Everything else will continue to work just fine.

**Comment by san7890** [comment]
yes but clone also should not be the "haha get damaged" type of damage for just raw murder (especially since it is equally trivial to fix with the right chems). there's no real good solution here sadly

**Comment by Exester509** [comment]
What about brain damage instead of a random organ? Damaging an organ rarely does anything

**Comment by san7890** [comment]
there's another heretic that already does brain damage iirc... so i remain unsure.

**Comment by distributivgesetz** [comment]
> there's another heretic that already does brain damage iirc... so i remain unsure.

Even though the amount of heretic paths we have is starting to hurt the variety between all of them, remember that cosmic heretics can mark people to deal combos and to burn people in their sight, which is already a unique mechanic. So the discussion should be if the damage dealt is balanced instead of the damage being unique to this path

**Comment by distributivgesetz** [comment]
I'm completely fine with axing the damage to any organ and bumping the burn damage up to the current total damage dealt by burn + clone, or changing this to only damage the brain, I just haven't decided if brain damage is a fitting punishment here or if it's too annoying anyways

**Review by Ghommie — APPROVED** [review]

**Pull request merged**